android/src/test/java/com/onegini/mobile/sdk/reactnative/ValidatePinWithPolicyUseCaseTests.kt

```diff
@@ -0,0 +1,79 @@
+package com.onegini.mobile.sdk.reactnative
+
+import com.facebook.react.bridge.Promise
+import com.onegini.mobile.sdk.android.handlers.OneginiPinValidationHandler
+import com.onegini.mobile.sdk.android.handlers.error.OneginiPinValidationError
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onegini.mobile.sdk.reactnative.clean.use_cases.ValidatePinWithPolicyUseCase
+import com.onegini.mobile.sdk.reactnative.exception.OneginiWrapperErrors
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Answers
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.any
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+@RunWith(MockitoJUnitRunner::class)
+class ValidatePinWithPolicyUseCaseTests {
+
+    @get:Rule
+    val reactArgumentsTestRule = ReactArgumentsTestRule()
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    lateinit var oneginiSdk: OneginiSDK
+
+    @Mock
+    lateinit var promiseMock: Promise
+
+    @Mock
+    lateinit var oneginiPinValidationErrorMock: OneginiPinValidationError
+
+    @Test
+    fun `When supplying null as pin should reject with PARAMETERS_NOT_CORRECT error`() {
+        val pin = null
+        Mockito.`when`(oneginiSdk.oneginiClient.userClient.authenticatedUserProfile).thenReturn(UserProfile("testId"))
+        ValidatePinWithPolicyUseCase(oneginiSdk)(pin, promiseMock)
+
+        verify(promiseMock).reject(OneginiWrapperErrors.PARAMETERS_NOT_CORRECT.code, "Expected parameter 'pin' to be String but was NULL")
+    }
+
+    @Test
+    fun `When pin is not null should call validatePinWithPolicy on the onegini sdk`() {
+        val pin = "14789"
+        ValidatePinWithPolicyUseCase(oneginiSdk)(pin, promiseMock)
+
+        verify(oneginiSdk.oneginiClient.userClient, times(1)).validatePinWithPolicy(any(), any())
```

Comment:
`times(1)` is the default mode, so you can just call `verify(oneginiSdk.oneginiClient.userClient)`.
You can also verify if the pin provided to the SDK is the same as the one passed to the use case. So instead of `any()` it would be somehting like `eq("14789".toCharArray())`